crates/iceberg/src/spec/schema.rs

```diff
@@ -643,6 +655,27 @@ pub(super) mod _serde {
         pub fields: StructType,
     }
 
+    /// Helper to serialize/deserializa Schema
+    impl TryFrom<SchemaEnum> for Schema {
+        type Error = Error;
+        fn try_from(value: SchemaEnum) -> Result<Self> {
+            match value {
+                SchemaEnum::V2(value) => value.try_into(),
+                SchemaEnum::V1(value) => value.try_into(),
+            }
+        }
+    }
+
+    impl From<Schema> for SchemaEnum {
+        fn from(value: Schema) -> Self {
+            if value.schema_id == 0 {
```

Comment:
Since the only difference between v1 and v2 for schema is `schema_id`, I think we should always convert to V2 since this is compatible with both formats. Using `schema_id == 0` may be incorrect since 0 maybe a valid schema  id .

Reply:
Thanks. I will update. Less code and much simpler i like it.
Moreover difference between V1 and V2 is very thin.